### PR TITLE
UTOPIA-433 & 435: Frontend Create Form UI & Integrate Frontend with API Endpoint

### PR DESCRIPTION
**Pull request by Bottle7**
_My apologies for the large PR_

# Description

This PR includes the following proposed change(s):

- Implements form UI from Figma designs
- Creates interfaces related to PIA Intake form and submission
- Adds _form.scss file
- Implements PIA Intake form submission POST request
- Creates Dropdown component

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

- Visually assessed styling
  - Firefox
  - Chrome
- Filled out form and submitted

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [x] New and existing unit tests pass locally with my changes
- [x] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

**Comment by akroon3r** [comment]
Why are we doing API and Frontend changes in the same PR?

**Comment by binggaofw** [comment]
@Bottle7  can you update the ppq page "Get started" button link fro PIA-intake, I put value /piaintake-form but I see you are using "pia-intake"

**Comment by binggaofw** [comment]
also in devTools page i can see this warning message
Screen Shot 2022-11-28 at 1 29 16 PM

**Comment by Bottle7** [comment]
> Why are we doing API and Frontend changes in the same PR?

We are not.

**Comment by binggaofw** [comment]
> Why are we doing API and Frontend changes in the same PR?

I think this pr only change frontend code

**Comment by akroon3r** [comment]
> > Why are we doing API and Frontend changes in the same PR?
> 
> We are not.

My fault, reading the title of the PR incorrectly.

**Review by binggaofw — APPROVED** [review]

File: src/frontend/src/components/common/InputText/InputText.tsx [hunk 1]
```diff
@@ -28,6 +38,15 @@ const InputText = ({
   return (
     <div className={'form-group ' + className}>
       <label htmlFor={inputId}>{label}</label>
+      {helperText !== '' && (
+        <p className="form-group__p--margin-zero">
+          {helperText}
+          <a href={linkHref} rel="noreferrer external" target="_blank">
+            {linkText}
+            {icon === true && <FontAwesomeIcon className="helper-text__link-icon" icon={faUpRightFromSquare} />}
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/common/InputText/InputText.tsx:46]: `icon` as a named prop sounds like if the component is accepting an icon object. Whereas your usage seems weather or not using an icon. I would advice renaming the prop to `showTargetIcon` or `hasIcon` types

Comment (kushal-arora-fw) [inline src/frontend/src/components/common/InputText/InputText.tsx:46]: `icon === true && ...` is equivalent to `icon && ...` or a safer `!!icon` if unsure of the type

**Comment by kushal-arora-fw** [comment]
Also, there's a weird navigation bar where PPQ is moved to almost centre.

image

**Comment by Bottle7** [comment]
> Also, there's a weird navigation bar where PPQ is moved to almost centre.
> 
> image

This occurred after a rebase. Not quite sure what changed.

**Comment by kushal-arora-fw** [comment]
docker build is failing with the following errors

image

**Comment by Bottle7** [comment]
> docker build is failing with the following errors
> 
> image

I just rebuilt the image 3 more times and got no errors. Also, I ran `docker system prune -f` between builds.

**Comment by binggaofw** [comment]
i can build the docker in my local

File: src/frontend/src/types/interfaces/pia-intake.interface.ts [hunk 2]
```diff
@@ -13,7 +13,8 @@ export interface IPIAIntake {
       mpoName: string,
       mpoEmail: string,
       initiativeDescription: string,
+      initiativeScope: string,
       dataElementsInvolved: string,
-      hasAddedPiDataElements: boolean | null,
-      riskMitigation: string | null,
+      hasAddedPiToDataElements: boolean,
+      riskMitigation: string | undefined,
 }
```
Comment (Bottle7) [inline src/frontend/src/types/interfaces/pia-intake.interface.ts:20]: Don't these have to accept null as a potential type as we are passing them in as null in some cases?

**Pull request merged**